blocklanguages.php

```diff
@@ -41,52 +41,12 @@ public function __construct()
 
 		$this->displayName = $this->l('Language selector block');
 		$this->description = $this->l('Adds a block allowing customers to select a language for your store\'s content.');
-		$this->ps_versions_compliancy = array('min' => '1.6', 'max' => _PS_VERSION_);
+		$this->ps_versions_compliancy = array('min' => '1.7', 'max' => _PS_VERSION_);
 	}
 
 	public function install()
 	{
-		return (parent::install() && $this->registerHook('displayNav') && $this->registerHook('displayHeader'));
-	}
-
-	protected function _prepareHook($params)
-	{
-		$languages = Language::getLanguages(true, $this->context->shop->id);
-		if (!count($languages))
-			return false;
-		$link = new Link();
-
-		if ((int)Configuration::get('PS_REWRITING_SETTINGS'))
-		{
-			$default_rewrite = array();
-			if (Dispatcher::getInstance()->getController() == 'product' && ($id_product = (int)Tools::getValue('id_product')))
-			{
-				$rewrite_infos = Product::getUrlRewriteInformations((int)$id_product);
-				foreach ($rewrite_infos as $infos)
-					$default_rewrite[$infos['id_lang']] = $link->getProductLink((int)$id_product, $infos['link_rewrite'], $infos['category_rewrite'], $infos['ean13'], (int)$infos['id_lang']);
-			}
-
-			if (Dispatcher::getInstance()->getController() == 'category' && ($id_category = (int)Tools::getValue('id_category')))
-			{
-				$rewrite_infos = Category::getUrlRewriteInformations((int)$id_category);
-				foreach ($rewrite_infos as $infos)
-					$default_rewrite[$infos['id_lang']] = $link->getCategoryLink((int)$id_category, $infos['link_rewrite'], $infos['id_lang']);
-			}
-
-			if (Dispatcher::getInstance()->getController() == 'cms' && (($id_cms = (int)Tools::getValue('id_cms')) || ($id_cms_category = (int)Tools::getValue('id_cms_category'))))
-			{
-				$rewrite_infos = (isset($id_cms) && !isset($id_cms_category)) ? CMS::getUrlRewriteInformations($id_cms) : CMSCategory::getUrlRewriteInformations($id_cms_category);
-				foreach ($rewrite_infos as $infos)
-				{
-					$arr_link = (isset($id_cms) && !isset($id_cms_category)) ?
-						$link->getCMSLink($id_cms, $infos['link_rewrite'], null, $infos['id_lang']) :
-						$link->getCMSCategoryLink($id_cms_category, $infos['link_rewrite'], $infos['id_lang']);
-					$default_rewrite[$infos['id_lang']] = $arr_link;
-				}
-			}
-			$this->smarty->assign('lang_rewrite_urls', $default_rewrite);
-		}
-		return true;
+		return (parent::install() && $this->registerHook('displayNav'));
```

Comment:
At the moment we kept the same hook structure for the top of the page as in default-bootstrap: displayBanner, displayNav, displayTop. It might still change depending on feedback!